### PR TITLE
@W-10459712@: Fifth wave of linting fixes.

### PR DESCRIPTION
**Pull request by jfeingold35**
Fixed linting issues in:
- `BaseEslintEngine.ts`
- `CustomEslintEngine.ts`
- `EslintCommons.ts`
- `JavascriptEslintStrategy.ts`
- `LWCEslintStrategy.ts`
- `TypescriptEslintStrategy.ts`
- `SarifFormatter.ts`
This required supplemental changes to:
- `package.json`
- `types.d.ts`
- `BaseEslintEngine.test.ts`
- `CustomEslintEngine.test.ts`
- `EslintTestDataGenerator.ts`
- `TypescriptEslintStrategy.test.ts`
- `yarn.lock`

### CODE REVIEW

File: src/lib/eslint/BaseEslintEngine.ts
Comment (jfeingold35) [inline src/lib/eslint/BaseEslintEngine.ts:28]: Simple alias lets us avoid using a cumbersome syntax elsewhere in the class.

File: package.json
Comment (jfeingold35) [inline package.json:63]: 3PP pending manual approval for some reason: https://gus.lightning.force.com/lightning/r/ADM_Third_Party_Software__c/a0qEE0000002RFRYA2/view

File: package.json
Comment (jfeingold35) [inline package.json:65]: `@types/eslint` uses this package to define some of its own types, and adding it ourselves lets us use it for matching definitions.
3PP pending manual approval for some reason: https://gus.lightning.force.com/lightning/r/ADM_Third_Party_Software__c/a0qEE0000002RH3YAM/view

File: src/lib/eslint/BaseEslintEngine.ts
Comment (jfeingold35) [inline src/lib/eslint/BaseEslintEngine.ts:43]: Renamed method to `getRunOptions`, to avoid confusion with the various Config-related types.

File: src/lib/eslint/BaseEslintEngine.ts
Comment (jfeingold35) [inline src/lib/eslint/BaseEslintEngine.ts:48]: `Strategy`s still implement this method, but it's only ever called within the classes themselves. So the method has been removed from the abstract class contract.

File: src/lib/eslint/BaseEslintEngine.ts
Comment (jbartolotta-sfdc) [inline src/lib/eslint/BaseEslintEngine.ts:284]: Is the idiomatic ts check in that case `!env`?

File: src/lib/eslint/EslintCommons.ts
Comment (jbartolotta-sfdc) [inline src/lib/eslint/EslintCommons.ts:68]: should this be `===`?

File: src/lib/eslint/BaseEslintEngine.ts
Comment (jfeingold35) [inline src/lib/eslint/BaseEslintEngine.ts:189]: The method has been altered to return the redefined `ESRuleConfig` type, and we immediately dereference its `rules` property. That lets us get around the need to import `Linter.RulesRecord` or define an alias for it.

File: src/lib/eslint/BaseEslintEngine.ts
Comment (jfeingold35) [inline src/lib/eslint/BaseEslintEngine.ts:218]: The various new/newly-imported types let us replace the soft references with hard ones.

File: src/lib/eslint/BaseEslintEngine.ts
Comment (jfeingold35) [inline src/lib/eslint/BaseEslintEngine.ts:234]: To satisfy the linter/compiler, we call the typeguard function before trying to assign the override to `config.baseConfig.env`.

**Review by jbartolotta-sfdc — APPROVED** [review]
One question, reminder that the target branch needs to be changed before merging.

File: src/lib/eslint/BaseEslintEngine.ts
Comment (jfeingold35) [inline src/lib/eslint/BaseEslintEngine.ts:240]: The `ESResult` type wasn't compatible with the `ESLint.LintResult` type, so I just switched from the former to the latter.

File: src/lib/eslint/BaseEslintEngine.ts
Comment (jfeingold35) [inline src/lib/eslint/BaseEslintEngine.ts:274]: As mentioned above, the method now returns the redefined `ESRuleConfig` type, and the results are stored in the `rules` property.

File: src/types.d.ts
Comment (jfeingold35) [inline src/types.d.ts:135]: These types didn't quite align with the `ESLint.LintResult` and `Linter.LintMessage[]` types, so I just swapped those types in for these, rendering these ones unnecessary.

File: src/lib/eslint/BaseEslintEngine.ts
Comment (jfeingold35) [reply src/lib/eslint/BaseEslintEngine.ts:284]: No. `!env` would return `true` if `env` is false, 0, or a few other things. `env == null` only returns true if it's `null` or `undefined`.

File: src/lib/eslint/EslintCommons.ts
Comment (jfeingold35) [reply src/lib/eslint/EslintCommons.ts:68]: Good catch.

**Pull request merged**
